kyuubi-server/src/test/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilderSuite.scala

```diff
@@ -17,16 +17,105 @@
 
 package org.apache.kyuubi.engine.flink
 
-import org.apache.kyuubi.KyuubiFunSuite
+import java.io.File
+import java.nio.file.{Files, Paths}
+import java.util.LinkedHashSet
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable.ArrayBuffer
+
+import org.apache.kyuubi.{FLINK_COMPILE_VERSION, KyuubiFunSuite, SCALA_COMPILE_VERSION}
 import org.apache.kyuubi.config.KyuubiConf
 
 class FlinkProcessBuilderSuite extends KyuubiFunSuite {
   private def conf = KyuubiConf().set("kyuubi.on", "off")
+  private def envDefault: Map[String, String] = Map(
+    "JAVA_HOME" -> s"${File.separator}jdk1.8.0_181")
+  private def confStr: String = {
+    val configBuffer = new ArrayBuffer[String]()
+    for ((k, v) <- conf.getAll) {
+      configBuffer += s"-D$k=$v"
+    }
+    configBuffer.toArray.mkString(" ")
+  }
+  private val javaPath = s"${envDefault("JAVA_HOME")}${File.separator}bin${File.separator}java"
+  private val flinkSqlClientJarPathSuffix = s"${File.separator}opt${File.separator}" +
+    s"flink-sql-client_$SCALA_COMPILE_VERSION-$FLINK_COMPILE_VERSION.jar"
+  private val flinkLibPathSuffix = s"${File.separator}lib${File.separator}*"
+  private val flinkConfPathSuffix = s"${File.separator}conf"
+  private val mainClassStr = "org.apache.kyuubi.engine.flink.FlinkSQLEngine"
 
   test("flink engine process builder") {
-    val builder = new FlinkProcessBuilder("vinoyang", conf)
-    val commands = builder.toString.split(' ')
-    assert(commands.exists(_ endsWith "flink-sql-engine.sh"))
+    val builder = new FlinkProcessBuilder("vinoyang", conf) {
+      override protected def env: Map[String, String] = envDefault +
+        ("HADOOP_CONF_DIR" -> s"${File.separator}hadoop${File.separator}conf") +
+        ("HBASE_CONF_DIR" -> s"${File.separator}hbase${File.separator}conf") +
+        ("YARN_CONF_DIR" -> s"${File.separator}yarn${File.separator}conf") +
+        ("HADOOP_CLASSPATH" -> s"${File.separator}hadoop")
+    }
+    val commands = builder.toString
+
+    val classpathEntries = new LinkedHashSet[String]
+    builder.mainResource.foreach(classpathEntries.add)
+    val flinkSqlClientJarPath = s"${builder.FLINK_HOME}$flinkSqlClientJarPathSuffix"
+    val flinkLibPath = s"${builder.FLINK_HOME}$flinkLibPathSuffix"
+    val flinkConfPath = s"${builder.FLINK_HOME}$flinkConfPathSuffix"
+    val hadoopConfPath = s"${File.separator}hadoop${File.separator}conf"
```

Comment:
Could this be optimized to iterate the above map for env?

Reply:
> Could this be optimized to iterate the above map for env?

ok